packages/core/src/new-api/internal/execution/execution-engine.ts

```diff
@@ -99,11 +116,22 @@ export class ExecutionEngine {
         };
       }
 
-      if (batchResult.some((b) => b.type === "execution-hold")) {
+      if (batchResult.some(isExecutionTimeout)) {
+        return {
+          status: "timeout",
```

Comment:
We need a new `timeout` execution result as that is a valid end result of a deploy now.